src/hssm/distribution_utils/dist.py

```diff
@@ -276,8 +332,24 @@ def dist(cls, **kwargs):  # pylint: disable=arguments-renamed
             return super().dist(dist_params, **other_kwargs)
 
         def logp(data, *dist_params):  # pylint: disable=E0213
+            if list_params[-1] == "p_outlier":
+                p_outlier = dist_params[-1]
+                dist_params = dist_params[:-1]
+
             logp = loglik(data, *dist_params)
 
+            if list_params[-1] == "p_outlier":
+                logp = pt.log(
+                    (1.0 - p_outlier) * pt.exp(logp)
+                    + p_outlier
+                    * pt.exp(
+                        pm.logp(
+                            get_distribution_from_prior(lapse).dist(**lapse.args),
```

Comment:
no efficiency concerns about this call got get_distribution_from_prior?

is this essentially a constructor and once the graph is built, this will not be "called" as such?

Reply:
That is correct. You can't use `pm.Distribution` directly without having a model in the context. That's why `bambi` has its own `bmb.Prior`. This basically uses `bmb.Prior` to construct the likelihood